src/components/pages/import-batch-view.js

```diff
@@ -1,24 +1,58 @@
-import React from 'react';
 import Page from '../page.js';
 import PropTypes from 'prop-types';
+import Logger from '../../common/logger.js';
 import { useTranslation } from 'react-i18next';
+import React, { useEffect, useState } from 'react';
+import { useService } from '../container-context.js';
 import { useDateFormat } from '../language-context.js';
+import { handleApiError } from '../../ui/error-helper.js';
 import { getImportedArticleUrl } from '../../utils/urls.js';
 import { DOCUMENT_IMPORT_TYPE } from '../../common/constants.js';
+import ImportApiClient from '../../services/import-api-client.js';
 import { Table, Row, Space, Tooltip, Collapse, List } from 'antd';
 import { importBatchDetailsShape } from '../../ui/default-prop-types.js';
-import { CloudDownloadOutlined, CloudSyncOutlined, WarningOutlined, CheckOutlined, ExclamationCircleOutlined, SyncOutlined } from '@ant-design/icons';
 import { isTaskSuccessful, taskStatusSorter, doesTaskHaveErrors } from '../../utils/task-utils.js';
+import { CloudDownloadOutlined, CloudSyncOutlined, WarningOutlined, CheckOutlined, ExclamationCircleOutlined, SyncOutlined } from '@ant-design/icons';
 
 const { Panel } = Collapse;
+const POLL_INTERVAL_IN_MS = 1000;
+const logger = new Logger(import.meta.url);
 
 function ImportBatchView({ initialState }) {
   const { t } = useTranslation('importBatchView');
   const { formatDate } = useDateFormat();
+  const importApiClient = useService(ImportApiClient);
+
+  const [batch, setBatch] = useState(initialState.batch);
 
-  const { batch } = initialState;
+  const id = batch._id;
+  const isCompletedImport = batch.progress === 1;
   const { hostName, allowUnsecure } = batch.batchParams;
 
+  useEffect(() => {
+    if (isCompletedImport) {
```

Comment:
I find this early return a bit counterintuitive here in this case, 
I'd suggest to turn it around and wrap the code block in `if (!isCompletedImport) { .. }` and avoid this fake cleanup return to just satisfy the linter.
Or just doing a `return;` and suppressing the linter, though I'd go for the first option personally.

Reply:
I wanted to avoid the unnecessary indentation, I'll do the first one.

Reply:
I went for the second option because the linter would complain anyway on the first option and we would get the indentation for nothing as well.